src/commands/timelinecommands.cpp

```diff
@@ -513,29 +513,32 @@ void MoveClipCommand::redo()
             }
         } else {
             // On redo, use the recorded indices to remove them.
-            trackIndex = clip.get_int(kTrackIndexProperty);
-            clipIndex = clip.get_int(kClipIndexProperty);
+            trackIndex = m_trackIndexList[i];
+            clipIndex = m_clipIndexList[i];
         }
         if (m_ripple)
             m_model.removeClip(trackIndex, clipIndex, m_rippleAllTracks);
         else
             m_model.liftClip(trackIndex, clipIndex);
+        ++i;
     }
     if (!m_redo) {
         m_selection = newSelection;
     }
     // Next, add the save clips to their new locations.
+    i = 0;
     for (auto &clip : m_selection) {
-        auto toTrack = clip.get_int(kNewTrackIndexProperty);
-        auto start = clip.get_int(kPlaylistStartProperty);
-        clip.set_in_and_out(clip.get_int(kShotcutInProperty), clip.get_int(kShotcutOutProperty));
+        auto toTrack = m_newTrackIndexList[i];
+        auto start = m_playlistStartList[i];
+        clip.set_in_and_out(m_inList[i], m_outList[i]);
```

Comment:
Maybe kPlaylistStartProperty and kTrackIndexProperty need to be set on the clip here. I am unsure.

Reply:
We can set them by passing values from `QList`s to producer properties.
But they won't be updated during undos because `MoveClipCommand::undo()` uses `UndoHelper`, and I don't see any lines in `UndoHelper` that contain `kPlaylistStartProperty` and `kTrackIndexProperty`.